packages/odyssey-react/src/components/List/List.tsx

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type {
+  ReactElement,
+  ComponentPropsWithRef,
+  ForwardedRef
+} from 'react';
+
+import {
+  forwardRefWithStatics,
+  useOmit,
+  useCx
+} from '../../utils';
+
+import styles from './List.module.scss';
+import ListItem from './ListItem';
+import DescriptionTerm from './DescriptionTerm';
+import DescriptionDetails from './DescriptionDetails';
+
+export type Props = {
+  /**
+  * List Items
+  */
+  children?: ReactElement | ReactElement[],
+  /**
+   * List element used 
+   */
+   listType?: 'unordered' | 'ordered' | 'description',
+   /**
+    * Remove default styling
+    */
+   unstyled?: boolean,
+} & ComponentPropsWithRef<'ul'> & ComponentPropsWithRef<'ol'> & ComponentPropsWithRef<'dl'>;
```

Comment:
Is this the correct typing? I think what we may want here is a descriminiated union where the props for say a `dl` aren't intersected with props for a `ul`, even if the component can render each of these based on the props passed in.